### ISSUE TITLE
Properties with dash defined with Flow aren't detected

### ISSUE THREAD

**Issue by FezVrasta** [yannickcr/eslint-plugin-react#1299]
```
type Props = {
  'data-action': string,
};

function Button({ 'data-action': dataAction }: Props) {
  return <div data-action={dataAction} />;
}
```

> error  'data-action' is missing in props validation  react/prop-types

**Comment by FezVrasta** [comment]
I noticed I can use a computed property name to avoid the error.

```
function Button({ ['data-action']: dataAction }: Props) {
```

**Comment by alexzherdev** [comment]
I made a test case with the code from the issue and it passed.

**Comment by ljharb** [comment]
Let’s merge that into master to close this issue :-)
